### PR TITLE
create a reusable search bar

### LINKED ISSUE #22: Create a reusable Search Bar Component
# Goal

Create a reusable Search Bar Component. 
**Note:** This should be a generic Search Bar which can be customized via props.

### PR DESCRIPTION
**Pull request by Grumpyyash**
This PR resolves #22 
As required, Created a reusable search bar component to search any items, projects, contributors or anything else. The component is highly customizable with headers, styles, methods and functions to be passed as props.

search

### CODE REVIEW

**Comment by Grumpyyash** [comment]
@wdevon99 I have separated the search bar component and will integrate it with project lists for searching soon which is issued in #23 . Meanwhile, could you please review this individual PR?

File: src/components/SearchBarComponent/index.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+                    onClick={() => handleSearch(searchItem)}
+                    className="icon"
+                    icon={faSearch} />
+
+                  </div>
+                 </div> :
+
+                 <div>
+                  <input
+                   className="search-input"
```
Comment (wdevon99) [inline src/components/SearchBarComponent/index.js:37]: Fix indentation

**Review by wdevon99 — APPROVED** [review]
LGTM

**Comment by Grumpyyash** [comment]
Already done in #176

**Pull request closed without merging**